src/libs/fileDownload/FileUtils.js

```diff
@@ -183,11 +186,11 @@ const readFileAsync = (path, fileName) =>
                 // For some reason, the File object on iOS does not have a uri property
                 // so images aren't uploaded correctly to the backend
                 file.uri = path;
-                resolve(file);
+                onSuccess(file);
             })
             .catch((e) => {
                 console.debug('[FileUtils] Could not read uploaded file', e);
-                resolve();
+                onFailure();
```

Comment:
What do you think about passing the thrown error `e` here as well? I can imagine the scenario where we would do something  in `onFailure` function based on the thrown error